neo/Prompt/Commands/SC.py

```diff
@@ -127,11 +132,83 @@ def command_desc(self):
         p6 = ParameterDesc('returntype', 'the returntype of the smart contract output')
         p7 = ParameterDesc('inputs', 'the test parameters fed to the smart contract, or use "--i" for prompted parameter input')
         p8 = ParameterDesc('--no-parse-addr', 'a flag to turn off address parsing when input into the smart contract', optional=True)
-        p9 = ParameterDesc('--from-addr', 'source address to take fee funds from (if not specified, take first address in wallet)\n\n'
+        p9 = ParameterDesc('--from-addr',
+                           'source address to take fee funds from (if not specified, take first address in wallet)\n\n'
                            f"{' ':>17} Usage Examples:\n"
                            f"{' ':>20} load_run path.py True False False 0710 05 input1 input2\n"
                            f"{' ':>20} load_run path.py True False False 0710 05 --i\n\n"
                            f"{' ':>17} For more information about parameter types see\n"
                            f"{' ':>17} https://neo-python.readthedocs.io/en/latest/data-types.html#contractparametertypes\n", optional=True)
         params = [p1, p2, p3, p4, p5, p6, p7, p8, p9]
         return CommandDesc('load_run', 'load a specified smart contract (.avm) file and test it', params=params)
+
+
+class CommandSCTestInvoke(CommandBase):
+    def __init__(self):
+        super().__init__()
+
+    def execute(self, arguments):
+        wallet = PromptData.Wallet
+        if not wallet:
+            print("Please open a wallet")
+            return False
+
+        arguments, from_addr = PromptUtils.get_from_addr(arguments)
+        arguments, invoke_attrs = PromptUtils.get_tx_attr_from_args(arguments)
+        arguments, owners = PromptUtils.get_owners_from_params(arguments)
+
+        if len(arguments) < 2:
+            print("Please specify the required parameters")
+            return False
+
+        hash_string = arguments[0]
+        try:
+            script_hash = UInt160.ParseString(hash_string)
+        except Exception:
+            # because UInt160 throws a generic exception. Should be fixed in the future
+            print("Invalid script hash")
+            return False
+
+        tx, fee, results, num_ops = TestInvokeContract(wallet, arguments, from_addr=from_addr, invoke_attrs=invoke_attrs, owners=owners)
+        if tx is not None and results is not None:
+
+            parameterized_results = [ContractParameter.ToParameter(item) for item in results]

```

Comment:
`parameterized_results` is only used in the debug string, maybe it would be better to call `ToJson()` in this comprehension.

It would give:
`parameterized_results = [ContractParameter.ToParameter(item).ToJson() for item in results]`

What do you think?

Reply:
sounds good